pkg/storage/plan_collaboratorStore.go

```diff
@@ -106,37 +108,33 @@ func (s *Store) PlanCollaboratorUpdate(
 
 // PlanCollaboratorDelete deletes the plan collaborator for a given id
 func (s *Store) PlanCollaboratorDelete(
+	tx *sqlx.Tx,
 	_ *zap.Logger,
 	id uuid.UUID,
 	userID uuid.UUID,
 ) (*models.PlanCollaborator, error) {
 
-	tx := s.db.MustBegin()
-	defer tx.Rollback()
-
+	// We need to set the session user variable so that the audit trigger knows who made the delete operation
 	err := setCurrentSessionUserVariable(tx, userID)
 	if err != nil {
 		return nil, err
 	}
 
-	stmt, err := tx.PrepareNamed(sqlqueries.PlanCollaborator.Delete)
-	if err != nil {
-		return nil, err
+	args := map[string]interface{}{
+		"id":          id,
+		"modified_by": userID,
 	}
-	defer stmt.Close()
 
-	collaborator := &models.PlanCollaborator{}
-	err = stmt.Get(collaborator, utilitysql.CreateIDQueryMap(id))
+	retCollaborator, err := sqlutils.SelectProcedure[models.PlanCollaborator](tx, sqlqueries.PlanCollaborator.Delete, args)
```

Comment:
I think this is supposed to be a get procedure. It's a delete by id, so we expect only one value to be returned, right?